configs/cache_partitioning/cache_partitioning.py

```diff
@@ -0,0 +1,218 @@
+        "max_capacity",
+    ], "Only support generating way and max_capacity policies"
+
+    if args.policy == "way":
+        allocated_ways = [way for way in range(0, args.way_allocation)]
+        allocation = WayPolicyAllocation(partition_id=0, ways=allocated_ways)
+
+        return WayPartitioningPolicy(allocations=[allocation])
+
+    if args.policy == "max_capacity":
+        return MaxCapacityPartitioningPolicy(
+            partition_ids=[0], capacities=[args.capacity_allocation]
+        )
+
+
+def configSystem():
+    """
+    Configure base system and memory
+    """
+
+    system = System(membus=IOXBar(width=128))
+    system.clk_domain = SrcClockDomain(
+        clock=STATIC_OPTIONS.system_frequency,
+        voltage_domain=VoltageDomain(),
+    )
+
+    # add memory
+    mem_range = AddrRange("128MB")
+    system.mem_ranges = [mem_range]
+
+    # Memory configuration
+    config_mem(STATIC_OPTIONS, system)
+    system.mem_ctrls[0].latency = STATIC_OPTIONS.request_latency
+    system.mem_ctrls[0].bandwidth = STATIC_OPTIONS.mem_bandwidth
+
+    return system
+
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+
+parser.add_argument(
+    "--policy",
+    default="way",
+    choices=["way", "max_capacity"],
+    help="This option defines which Cache Partitioning Policy to use for "
+    "the system cache",
+)
+
+parser.add_argument(
+    "--capacity-allocation",
+    type=capacityAllocation,
+    default=0.5,
+    help="The amount of the cache to partition to the default PartitionID "
+    "when using Max Capacity Cache Partitioning Policy in [0,1] range",
+)
+
+parser.add_argument(
+    "--way-allocation",
+    type=wayAllocation,
+    default=4,
+    help="The number of ways in the cache to partition to the default "
+    "PartitionID when using Way Cache Partitioning Policy",
+)
+
+args = parser.parse_args()
+
+
+m5.ticks.setGlobalFrequency(STATIC_OPTIONS.system_frequency)
+system = configSystem()
+
+# create a cache to sit between the memory and traffic gen to enforce
+# partitioning policies
+partition_policy = generatePartPolicy(args)
+system.cache = NoncoherentCache(
+    size="64KiB",
+    assoc=8,
+    partitioning_policies=[partition_policy],
+    tag_latency=0,
+    data_latency=0,
+    response_latency=0,
+    mshrs=1,
+    tgts_per_mshr=12,
+    write_buffers=1,
+    replacement_policy=MRURP(),
+)
+system.cache.mem_side = system.membus.cpu_side_ports
+
+# instantiate traffic gen with correct config and connect to crossbar
+tgen_config_path = os.path.join(SCRIPT_DIR, "tgen.cfg")
+system.tgen = TrafficGen(config_file=tgen_config_path)
+system.tgen.port = system.cache.cpu_side
+
+# finalise config and run simulation
+root = Root(full_system=False, system=system)
+root.system.mem_mode = "timing"
+m5.instantiate()
+
+# handle exit reporting
+exit_event = m5.simulate(STATIC_OPTIONS.max_ticks)
+print("\nExiting @ tick %i because %s" % (m5.curTick(), exit_event.getCause()))
```

Comment:
Please use f-strings.

```suggestion
print(f"Exiting @ tick {m5.curTick()} because {exit_event.getCause()}")
```

However, does this really make sense in this file? Do you want to report something else like the hit/miss ratio?

Reply:
Updated the PR to use f-strings;
I'm not sure whether there is elegant way to read the simulation stats directly from the python script; Did you have anything in mind for reporting the hit/miss ratio?

Reply:
This does exist, but it's not well documented (in fact there's no examples I can find). Don't worry about this for now :)

For your knowledge, it's the `gem5stats` module: `from m5.stats.gem5stats import get_simstat`